File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-1.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:


        while len(tokens) > 1:

            for i in range(len(tokens)):

                if tokens[i] in "+-/*":

                    a = tokens[i-2]
                    b = tokens[i-1]
                    res = 0
                    if tokens[i] == '+':
                        res = int(a)+int(b)

                    elif tokens[i] == '-':
                        res = int(a) - int(b)

                        
                    elif tokens[i] == '*':
                        res = int(a) * int(b)

                    elif tokens[i] == '/':
                        res = int(int(a)/int(b))
                    
                    tokens = tokens[:i-2] + [str(res)] + tokens[i+1:]
                    break
        
        return int(tokens[0])
                

        

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-1.java
class Solution {
    public int characterReplacement(String s, int k) {
        int n=s.length();
       Map<Character,Integer> mp=new HashMap<>();
       int count=0;
       int res=0;
       int start=0;
       int index=0;
       for(int i=0;i<n;i++)
       {
          mp.put(s.charAt(i),mp.getOrDefault(s.charAt(i),0)+1);
         
            count=Math.max(mp.get(s.charAt(i)),count);
         

          if(count+k<i-start+1)
          {
            mp.put(s.charAt(start),mp.get(s.charAt(start))-1);
            
            start++;
          }
          else
            res=Math.max(res,i-start+1);
       }
       return res;
        
    }
}
